jobs/delorean/jenkinsfiles/branch/suite-next/Jenkinsfile

```diff
@@ -0,0 +1,112 @@
+#!groovy
+
+// Ensure required parameters are provided
+def verifyParameters(suiteParams) {
+  if (!suiteParams.gitUrl || !suiteParams.suiteBranch) {
+      def userInput = input message: 'Suites Next Options', parameters: [
+        string(defaultValue: (suiteParams.gitUrl ?: 'git@github.com:integr8ly/installation.git'), description: 'The installation repo git url', name: 'installationGitUrl'),
+        string(defaultValue: (suiteParams.suiteBranch ?: ''), description: 'Desired name for the suite branch', name: 'suiteBranch'),
+      ]
+      suiteParams.gitUrl = userInput.installationGitUrl
+      suiteParams.suiteBranch = userInput.suiteBranch
+      verifyParameters(suiteParams)
+  }
+}
+
+// Checkout remote repository
+def checkoutGitRepo(gitUrl, gitRef, credentialsID) {
+  checkout([$class: 'GitSCM', branches: [[name: gitRef]],
+            doGenerateSubmoduleConfigurations: false,
+            extensions: [],
+            submoduleCfg: [],
+            userRemoteConfigs: [[credentialsId: credentialsID, url: gitUrl]]])
+}
+
+// Get product next branches available in the installation repo
+def getProductNextBranches(gitOrg, gitRepo, sourceBranch) {
+  nextBranches = sh(returnStdout: true, script: "git ls-remote origin | grep next | awk '{print \$2}'").replaceAll("refs/heads/", "").split()
+
+  // Do not include the source branch in the next branches list
+  productNextBranches = nextBranches.minus(sourceBranch)
+  return productNextBranches
+}
+
+def gitCreateAndCheckoutBranch(branchName, pushOnCreate = false) {
+  sh "git checkout -b ${branchName}"
+
+  if (pushOnCreate) {
+    sh "git push origin ${branchName}"
+  }
+}
+
+def gitDeleteBranch(branchName) {
+  String remoteBranchCommit = sh(returnStdout: true, script: "git ls-remote origin refs/heads/${branchName} | cut -f 1").trim()
+  if (remoteBranchCommit) {
+    sh "git push origin --delete ${branchName}"
+  }
+}
+
+def suiteParams = [:]
+def ghOwner = "integr8ly"
+def ghRepo = 'installation'
+def githubSSHCredentialsID = 'jenkinsgithub'
+def masterBranch = 'master'
+def suiteBranchList = ['integreatly-next', 'integration-next']
```

Comment:
This is fine for now, but as we add more suite types we might find it more useful to just name the branches appropriately as suite types e.g suite-integreatly-next, suite-integeration-next